### DISCUSSION TITLE
1/8th scale mechanisms

### DISCUSSION THREAD

**Discussion by ProPeach** [tsunamayo/Starship-EVO#3955] [Suggestions]
Love this new forum!

Building in novox, it feels very strange that while the vast majority of bricks can be scaled down to 1/8th, mechanisms like Hinges or Rotors cannot. When working in 1/8th scale for hatches, doors or flaps a hinge 8 times larger than your other blocks is pretty unwieldy, even when partially hidden.

This is one example of many

Here's with a 1/4 scale hinge-
![image](https://user-images.githubusercontent.com/27857711/124797759-8c56ba00-df4a-11eb-9723-9fbe5bb71032.png)

Whereas this would be what a 1/8 scale hinge looks like -
![image](https://user-images.githubusercontent.com/27857711/124798277-14d55a80-df4b-11eb-80e0-9cb81da335e6.png)

Unless there's some gameplay limitation you have in mind which would be super interesting to hear about, I think reducing the number of bricks which scale to different sizes would be a lot more consistent and less confusing for a newer player.

**Comment by AsageSoftworks** [comment]
Tsuna,

I agree with peach. But I also souod like to suggest thr same for other devices. Systems(power, trhust, etc.) Also need larger versuons past the 4m scale, and 1/8th scale main thrusters would be amazing!

Thank you for your amazing work on this glorious game!
A Sage

**Reply by ProPeach** [reply]
I'm actually totally ok with the larger scale of the Lightcruise thrusters, they're meant to be a little larger and heavier to be suited for larger ships capable of lightcruise. Maybe that's just me thinking in old block vs brick terms though.

On the other hand, Lightcruise thrusters larger than 4m scale would be fantastic, if not just to remove the borders you get when you have to place a couple of them together on larger ships

**Reply by Gliese-832-c** [reply]
I fully agree with you. We need larger systems for the biggest of ships. I also agree on not having small main thrusters, because a tiny little fighter isn't supposed to be able to do everything a bigger ship can do.

**Reply by SnowyBirdot** [reply]
> I fully agree with you. We need larger systems for the biggest of ships. I also agree on not having small main thrusters, because a tiny little fighter isn't supposed to be able to do everything a bigger ship can do.

Plus, it's fully feasible to make a very small, sleek, and lighcruise capable fighter, it just takes sacrifices in other areas. But honestly, if the fighter is too small for a 1x1x1 at least. It may just be too small.

**Comment by Gliese-832-c** [comment]
Totally agree. While we're at it, how about larger systems for truly gargantuan capital ships/space stations?

**Comment by Xerma** [comment]
Although I totally agree that we should have 1/8th mechanisms because we have 32m sized versions, I'm not so sure we need bigger systems

**Reply by Garrett-C** [reply]
Why wouldn't we want bigger systems. Sounds perfect for bigger ships.

**Reply by Xerma** [reply]
With the current scale of 4m, it doesn't take all that much to bring a 9km ship up to full capacity in terms of systems. I just think that if systems scaled all the way up to 32m then it would, in a way, promote absolutely massive ships > more room for this and that > bigger ships are going to need more turrets or more support ships to protect themselves which is going to add quite a bit more lag. 

People can for sure just have multitudes of the 4m systems all about their massive ships anyways but I do think allowing systems to scale past a point promotes the idea that it must be ok to build gargantuan things. I see 4m being the max scale for systems to be a good soft limit, while not actually being limiting.

**Reply by Garrett-C** [reply]
I mean surely the point of the 32m scale in general is to promote that large builds are ok... this would just mean the number of blocks needed at that scale would be lower. Rather than higher with keeping systems so limited. Seems to contradict the large hull blocks to have small systems.

**Reply by Xerma** [reply]
Sure the number of systems needed would be lower, however, in this game you don't need gigantic amounts of systems to get ships up and running - that's kind of the whole point. 

This is a total of 26 blocks @ 10% heat production of a nearly 10km ship so I hardly think upscaling systems will reduce lag in terms of block count. But I do believe people will want to try and take advantage of having much larger systems to make much much much larger builds that will require even more child entities for various functions: elevators, large doors, railways to get from A to B, and especially turrets. 
![20210718133039_1](https://user-images.githubusercontent.com/32974671/126078831-c18c1fdd-9d23-4006-a430-513c66b7144c.jpg)

I did a test a few days ago to see how many moving child entities the game can handle concurrently, this image is with 144 child entities moving all in sync with 25fps:
![20210714092329_1](https://user-images.githubusercontent.com/32974671/126078892-87816dd7-6d03-45e1-8315-2670996295b8.jpg)

If people strictly used 2 child entities for 1 turret, which some use more than 2, then that would amount to 72 turrets. Now a single large ship is unlikely to have that many but do consider that dropped to 25fps with a GTX 970. With GPU prices being as they are today it is not terribly likely to rely on the higher end cards to be the standard so I would like to assume that a card of my caliber is more likely the average occurrence, if not slightly below the average. 

So from that test and my working on a nearly 10km ship I don't buy that we need systems to scale to 32m. Especially not if we want to have larger scaled battles. Now maybe turret spam/child entities could get optimized more, that part I don't know. In my opinion I could only suggest maybe bumping their max scale to 8m from 4m. 

A separate suggestion would be to have a new mechanism that would allow for rotation on 2 axes so as to cut down on the child entities needed to make turrets

**Reply by Garrett-C** [reply]
That's sort of the thing though. Large blocks make that possible anyway and people will still build monster ships but now instead of having that monster ship with all those things that add to the performance impact it will also have a greater number of system blocks. 

So your solution just makes the problem worse.

**Reply by cloudrambler** [reply]
There is a common theme in real world human engineering that we make functional parts to sizes that are manageable to the workforce. For example, the engine of a car is something like 20% the size of the vehicle, but the engine of a large container ship, whilst much bigger than the car engine directly, is a far smaller proportion of the vehicle size. This is because scale brings efficiency and the power output of an engine scales faster with size than the power consumption of the vehicle. So we don't need the engines to be so large. Even if we do, we still choose to build multiple large engines instead of a single giant one, because its easier and safer for us to make and work on. It's also more cost effective and reliable to replicate larger numbers of smaller system that can be individually repaired etc.

So I think for any human(oid) made starship, the functional parts would have a scale range that is smaller than the ship scale range. If we would need to cram a huge ship with power hungry systems, then we would realistically use common efficient power systems and duplicate them as required.

The only exception to this philosophy would be for ships that are not of human(oid) origin. Would a self-replicating machine species keep to a finite size limit or would they scale linearly, like fishes do. So maybe a species of space faring fishes would have identically designed ships of literally different scales to suit the occupants.

Generally I agree with @Xerma that a system size limit lower than the block size limit is both workable and representative of real world engineering methods.

**Reply by Garrett-C** [reply]
See the thing to remember is we aren't talking about real life, we are talking about a game so the focus of designing these systems should be on gameplay not on what we do for boats in real life. 

Currently, we have blocks that have a fixed scale, like buttons and logic, blocks that can be sized anywhere between 1/8m and 32m like hull, and some blocks that have their own unique ranges like the systems. This isn't intuitive from a building perspective. 

Secondly when people build ships with 32m blocks and are forced to use systems that scale to 4m the build is equivalent to using 1/8 scale systems with 1m blocks. It's not all that efficient or an enjoyable experience. It's actually less so than using exclusively brick systems in the old versions of the game. 

The idea of system scaling is that larger systems are more efficient so not sure what your argument about them being power-hungry is really meant to say. The systems are more power-hungry by making them smaller.

I think the flaw in your way of thinking is that the focus should be on player experience, not on real-world engineering. The game has always positioned itself as being more arcady than realistic so hurting gameplay for realism like that seems odd to me.

**Reply by cloudrambler** [reply]
Of course what we build in STEVO is not limited by what we do now on Earth. I'm just pointing out the natural human tendency to make things at sizes which we can physically interact with.

You have to imagine what it would be like to walk through these ships. To me, it would feel more immersive to have an endless chamber filled with house sized power units (each with a control console) than it would to see a one Km cube of fairly featureless dark box. So I am suggesting that what we do now, and what we find intuitively interesting, could turn out to be the same thing.

I get the point about 4m system blocks being too small for a ship bade entirely of 32m hull blocks, but would you actually build that way, using JUST the largest blocks? Lets say you want a 10Km ship (which is freakin enormous from a playability standpoint) and you want it to have a nice shape. Building with only 32m block slopes at an average ratio of 1:8, means you only have room for 39 blocks over the length. I would find that limiting and choose to subdivide the blocks to 16m, 8m, 4m, etc. to make use of the variable proportions that those scales provide.

My point is that building from only 32m blocks is no more sensible than building from only 1/8m blocks. IMO, the useful layout scales are from 1m to 4m. Anything smaller than 1m is for aesthetic details and anything larger than 4m is for reducing block count for larger flat surfaces. Building with only 32m blocks would be large for it's own sake and not interesting to work inside.

**Reply by Garrett-C** [reply]
Having larger systems as an option for the player does not prevent you building immersive rooms with smaller systems. It does however allow you to build the bulk of your ships performance in a non tedious way. 

As I expressed filling a ship made mainly from 32m blocks with 4m systems is half as fast as filling a ship made from mainly 1m blocks with .25m systems. It's just not enjoyable from a gameplay perspective. 

Walking through these ships is one thing but building them is another thing. The player experience of adding systems like that is not going to be fun. And if building them isn't fun it's unlikely you will be walking through them, you know because nobody wants to build them. 

What's the difference between building a 16000m ship with mainly 32m blocks and building a 500m ship with mainly 1m blocks? Yeah I am not saying you would only use tbe biggest size available, of course you would use smaller sizes for detail. However 32m blocks allow us rp build massive ships. The systems do not match that. 

My point is that it doesn't matter what scale you or I build with someone will build at a different scale and currently the system size cap just makes the game more tedious at those larger scales. Again I suggest you try building a 500m ship and filling it with 1/8 systems and tell me how much fun that is.

Additionally remember that systems are more efficient at larger scales so contrary to your real world example as you get bigger system blocks would need to account for more and more of the ships interior.

**Reply by Gliese-832-c** [reply]
I totally agree with Garret. So, the amount of child entities that lag the game was mentioned. But if you think about it, wouldn't it be better to have 10 giant turrets than 100 small ones?

**Reply by Xenoprophet-Nail** [reply]
small pistons, hinges, and rotators would allow for some rather neat doors and prefab furniture. I'm having issues with building hovercars in detail because there's no easy way to have it so the windows roll up and down, have to use a 1/4 hinge for a 1/8th window and you can see how making that is a nightmare. Smaller bits for handhelds and the upcoming mechs would allow for more creative details. You have to remember some of us don't build on massive scales.

**Reply by SnowyBirdot** [reply]
That's definitely something important to consider, for instance my gig isn't really making huge ships, it's making feature dense ones, which this would be a godsend for.

**Comment by TIKIRobo** [comment]
it would also be cool to get 1/8 scale buttons and switches as 25cm buttons are kinda big

**Reply by Xenoprophet-Nail** [reply]
I have to agree Tiki smaller switches and buttons means more variations in cockpit designs.

**Reply by Gliese-832-c** [reply]
How about getting bigger buttons and switches as well? I know the use would be rather situational, but I still think that it would be cool to have a "big red button" that you press to enable some big system like a giant door or weapons.

**Reply by SnowyBirdot** [reply]
I mean, it is already a really, really big button. But yeah I can't say that the idea of a like, meter wide button, doesn't make me giggle a little.

**Reply by Gliese-832-c** [reply]
I just had an idea: What if buttons could be triggered by colliding ships? Then you could have 8m+ buttons that get triggered by ships flying into them. That would be really cool!

**Comment by SnowyBirdot** [comment]
I have been dreaming of this very thing, so I totally support it, also yeah this is much better for suggestions.

**Comment by ProPeach** [comment]
Hey @tsunamayo! Thanks for answering the question concerning 1/8th mechanisms in the recently Q&A. Can you go into a little more detail on what you think of them here? From an outside perspective, it's easy to think they're just the same as existing mechanisms but with a smaller stretch limit which wouldn't change much. Are there any particular technical challenges they might introduce, like collisions etc? Or are you concerned about overuse?
